benchmark_utils/latent_signature_utils.py

```diff
@@ -101,34 +101,36 @@ def create_latent_signature(
                     )
                     adata_sampled = adata[sampled_cells]
 
-                if signature_type == "pre-encoded":
-                    assert (
-                        model is not None,
-                        "If representing a purified pseudo bulk (aggregate before embedding",
-                        "), must give a model",
-                    )
-                    assert (
-                        count_key is not None
-                    ), "Must give a count key if aggregating before embedding."
-                    
-                    if use_mixupvi:
-                        result = model.get_latent_representation(
-                            adata_sampled, get_pseudobulk=True
-                        ).reshape(-1)
-                    else:
+                assert (
+                    model is not None,
+                    "If representing a purified pseudo bulk (aggregate before embedding",
+                    "), must give a model",
+                )
+                assert (
+                    count_key is not None
+                ), "Must give a count key if aggregating before embedding."
+                
+                if use_mixupvi:
+                    # TODO: in this case, n_cells sampled will be equal to self.n_cells_per_pseudobulk by mixupvae
```

Comment:
Hum no this is more complicated than that, and one of the reasons I'm saying the pipeline should be reformatted. 
Normally, we would create our own purified pseudobulk to then create the latent signature by taking the latent z of these purified pseudobulks. But when using categorical and continuous covariates during training, taking z during inference didn't average them the way it was supposed to do (thus we have a flawed representation). To counter that in the case of mixupvi, I originally added the possibility to take z_pseudobulk during inference directly instead of z (that's what line 116 returns). But I realised during this PR that z_pseudobulk only averages on `self.n_cells_per_pseudobulk` cells, instead of all cells or sc_per_pseudobulk, like we wanted in this function. Si I disabled the possibility and added the comment.
I hope this is clear...

But looking again at the code, I also realised that there is an issue at line 128, because the argument get_pseudobulk only exists in the case of MixUpVI. I will change that in a new commit.